### ISSUE TITLE
how to modify in_features of classifier in vgg 

### ISSUE THREAD

**Issue by Saharkakavand** [mortezamg63/Accessing-and-modifying-different-layers-of-a-pretrained-model-in-pytorch#2]
I pruned some of the filters in last conv layer, now I need to change 512*7*7 of linear layer, how can I do it?

**Comment by mmoha014** [comment]
Your question is not clear for me. But If your want to change it it does not seem correct.

This property is in fully connected layer. it is better to change it using the following code.

model_ft = models.resnet18(pretrained=True)
num_ftrs = model_ft.fc.in_features
model_ft.fc = nn.Linear(num_ftrs, 2) -> here you can change number of input and output

when you want to change a layer, it's recommended to define the layer again.

I hope to understand your question.

**Comment by Saharkakavand** [comment]
I trained the vgg and saved the model as pth file. then I load it for pruning some filters of it.
the last conv after pruning is not 512 anymore, some filters are gone.( I dont know the mapping between conv layer and linear layer)
how Pruning the last conv layer affects the first linear layer of the classifier which is (512 *7 *7, 4096).
how can I prune the input weights of classifier according to the last conv layer.

`class VGGOWN(nn.Module):

def __init__(self):
    super(VGGOWN, self).__init__()
    self.features = nn.Sequential(
        nn.Conv2d(3, 64, 3, padding=1),
        nn.ReLU(inplace=True),
        nn.Conv2d(64, 64, 3, padding=1),
        nn.ReLU(inplace=True),
        nn.MaxPool2d(2, stride=2, ceil_mode=True),  # 1/2

        # conv2
        nn.Conv2d(64, 128, 3, padding=1),
        nn.ReLU(inplace=True),
        nn.Conv2d(128, 128, 3, padding=1),
        nn.ReLU(inplace=True),
        nn.MaxPool2d(2, stride=2, ceil_mode=True),  # 1/4

        # conv3
        nn.Conv2d(128, 256, 3, padding=1),
        nn.ReLU(inplace=True),
        nn.Conv2d(256, 256, 3, padding=1),
        nn.ReLU(inplace=True),
        nn.Conv2d(256, 256, 3, padding=1),
        nn.ReLU(inplace=True),
        nn.MaxPool2d(2, stride=2, ceil_mode=True), # 1/8

        # conv4
        nn.Conv2d(256, 512, 3, padding=1),
        nn.ReLU(inplace=True),
        nn.Conv2d(512, 512, 3, padding=1),
        nn.ReLU(inplace=True),
        nn.Conv2d(512, 512, 3, padding=1),
        nn.ReLU(inplace=True),
        nn.MaxPool2d(2, stride=2, ceil_mode=True),  # 1/16

        # conv5
        nn.Conv2d(512, 512, 3, padding=1),
        nn.ReLU(inplace=True),
        nn.Conv2d(512, 512, 3, padding=1),
        nn.ReLU(inplace=True),
        nn.Conv2d(512, 512, 3, padding=1),
        nn.ReLU(inplace=True),
        nn.MaxPool2d(2, stride=2, ceil_mode=True) # 1/32
    )
    self.avgpool = nn.AdaptiveAvgPool2d((7, 7))

    self.classifier = nn.Sequential(

        nn.Linear(in_features=512 * 7 * 7, out_features=4096, bias=True),
        nn.ReLU(inplace=True),
        nn.Dropout(p=0.5),
        nn.Linear(in_features=4096, out_features=4096, bias=True),
        nn.ReLU(inplace=True),
        nn.Dropout(p=0.5),
        nn.Linear(in_features=4096, out_features=1000, bias=True)
    )

def forward(self, *input):
    x = self.features(input)
    #print(x.size)
    x = self.avgpool(x)
    #print(x.size)
    x = x.view(x.size(0), -1)
    #print(x.size)
    y = self.classifier(x)
    return y`

**Comment by mmoha014** [comment]
If I've understood the issue, I think you should measure output of each layer. When you change the number of filters, the number of outputs will be change. That's why it is not the same and you face error. There are some equations, which helps you find the output size after each convolutional layer. Read these slides to see the issue.
Sorry for late response. I am busy and sometime I can not answer. This is my another account. in GitHub Hope the guide works for  you.

https://drive.google.com/file/d/0B21Wl06uvFPFMmRPU05EZ1RQTTA/view

**Comment by Saharkakavand** [comment]
I found a solution of my problem. 
I know the output size of all layers I have them in an orderdict ( when I prune some filters I update the orderdict accordingly)
thanks for your consideration
